### ISSUE TITLE
gRPC service failed with status: TIMEOUT

### ISSUE THREAD

**Issue by AdriMsM91** [hashgraph/hedera-sdk-js#1411]
### Description

Each **50 of 3000 transactions** more or less , we are getting this error:
```js
error: GrpcServiceError: gRPC service failed with status: TIMEOUT
```

It happens either sending NFTs, either sending Hbar or inclusive creating a new Account. We are executing it in BackEnd in Node.js .

This is a big problem for us and we're having big troubles with our Clients because of this.

### Steps to reproduce

1-  .addNftTransfer()
2- .addHbarTransfer()
3-  Generating a new Account in Hedera 

### Hedera network

mainnet

### Version

2.19.1

### Operating system

None

**Comment by ochikov** [comment]
Hello @AdriMsM91, There is a default value of the `maxExecutionTime` of the Node Client in the SDK. The default time is set to 10 seconds. When the transaction is too big, you might hit the TIMEOUT error from the GRPC. You can use the setter: `setMaxExecutionTime` to increase this time and try again. You can also send us some code in order to test and replicate your issue.

**Comment by AdriMsM91** [comment]
Hello @ochikov the Transaction isn't too big is just an NFT Transfer. I know I can retry... but I shouldn't be worried everytime I'm using the SDK to add Works Arounds... As you can see there are more people with this same problem in Hedera Dicord #javascript 

```js
 let response = new TransferTransaction()
  .setMaxTransactionFee(new Hbar(maxFee))
  .addNftTransfer(tokenId, serialNumber, originAccount, receiverAccount)
  .setTransactionMemo(memo);
const txSubmit = await response.execute(client);
const receipt = await txSubmit.getReceipt(client);
```
```js
const newAccount = await new AccountCreateTransaction()
 .setKey(newAccountPublicKey)
 .execute(client);
const getReceipt = await newAccount.getReceipt(client);
```

**Comment by Burstall** [comment]
This is a behaviour I am seeing as of today (post the upgrade yesterday as never happened before):

Using: @hashgraph/sdk@2.19.2


GrpcServiceError: gRPC service failed with status: TIMEOUT
    at Timeout._onTimeout (...\@hashgraph\sdk\lib\channel\NodeChannel.cjs:111:20)
    at listOnTimeout (node:internal/timers:564:17)
    at process.processTimers (node:internal/timers:507:7) {
  status: GrpcStatus { _code: 17 }
}

calling smart contract methods (any even lightweight getters).

**Comment by ochikov** [comment]
We are going to investigate it. Keep in mind that the issue might be in some of the Nodes, and nodes not being able to respond. Can you confirm that this is through Node client or Web client? I need to know which type of GRPCs are accessed?

**Comment by Burstall** [comment]
node.js for me

If i retry it will work - worst I have had to do 2 retries to get it through.

**Comment by AdriMsM91** [comment]
> We are going to investigate it. Keep in mind that the issue might be in some of the Nodes, and nodes not being able to respond. Can you confirm that this is through Node client or Web client? I need to know which type of GRPCs are accessed.

Node.js for me also.

**Comment by ochikov** [comment]
Can you give me information about the node that you are trying to process through when the TIMEOUT is received?

**Comment by Burstall** [comment]
I was not pinning it down to a single node. just instantiated via Client.forTestnet()

**Comment by AdriMsM91** [comment]
> I was not pinning it down to a single node. just instantiated via Client.forTestnet()

Same but in my case  `Client.forNetwork('mainnet')`;

**Comment by ochikov** [comment]
There appears to have been some sort of brief GCP disruption, though they haven't reported any interruptions. But we did observe proxy checks on GCP proxies failing briefly around ~1100UTC, which lines up with when you posted. There doesn't appear to have been any impact on any actual nodes running in GCP.
This is what I've received from the Network team.

**Comment by AdriMsM91** [comment]
We have with this problems for almost 2 weeks, but we have detected is happening way more with v2.19.2

**Comment by AdriMsM91** [comment]
Good morning,

Yesterday after we launched 3 more Launchpads (One at 20 UTC) and 2 more at (23 UTC) , in 150 NFTs sent, we got 7 times `TIMEOUT`.

**Comment by ochikov** [comment]
Version 2.19.2 did not change anything related to the transaction workflow and timeouts. I am investigating it at the moment. From Yesterday, 1000 NFT transfers and 1000 HBAR transfers are successfully processed.

**Comment by ochikov** [comment]
I think that I've managed to isolate the issue. It is happening only with Node 10 with IP: 179.190.33.184

**Comment by ochikov** [comment]
@AdriMsM91 I've located the issue. Will work to be fixed. Thank you in advance for reporting it. 🙏

**Comment by AdriMsM91** [comment]
@ochikov Glad to hear that! Thanks you! <3

**Comment by ochikov** [comment]
Just to let you know how we are going to fix it: 
Let's say that you are hitting Node 10 which contains 3 different IPs. If one of the IPs is not responsible and we are not getting any error (timeout), we will wait some default time and then try to process the transaction through another Node. You now see the Timeout error because we are not marking this node as unhealthy. Now we will mark it, your execution will be awaiting to switch for a new node and try to process the transaction, BUT this transaction will not be discarded, but trying to process it. I will give you a beta release version when we are ready. Roughly estimated time for this fix will be tonight/tomorrow morning.

**Comment by ochikov** [comment]
[v2.20.0-beta.3](https://github.com/hashgraph/hedera-sdk-js/tree/release/v2.20.0-beta.3) Should fix the issue.

**Comment by mshakeg** [comment]
Just wanted to note that the `maxExecutionTime` passed to `setMaxExecutionTime` should be specified in ms not seconds.

**Comment by ochikov** [comment]
@mshakeg Yes, it is in milliseconds. All the timings in the SDKs are in milliseconds.
